recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/python3
+from optparse import OptionParser
+import sys
+
+force_update=False
+
+def sha256_file(file,length):
+    """
+    Return sha256 of the file
+    """
+    import mmap
+    import hashlib
+    method=hashlib.sha256()
+    try:
+        with mmap.mmap(file.fileno(),length,access=mmap.ACCESS_READ) as mm:
+            chunk = mm.read(length)
+            method.update(chunk)
+    except:
+         pass
+    return method.hexdigest()
+
+class FirmwareUpdate(object):
+    def __init__(self,force,erase):
+        self.force_update=force
+        self.erase_env=erase
+
+    def sha256_check(self,file):
+        """
+            Firmware Sha256 Check
+        """
+        check_sha256_file=file+'.sha256'
+        
+        try:
+            with open(check_sha256_file, 'r') as f:
+                check_sha256=f.read()
+        except:
+            raise FirmwareUpdateException("sha256 file not exists")
+
+        try:
+            with open(file, 'rb+') as f:
+                hash_code=sha256_file(f,0)
+        except:
+            raise FirmwareUpdateException("Firmware not Exists")
+        
+        if check_sha256 != hash_code:
+            if( self.force_update != True):
+                raise FirmwareUpdateException("File Checksum is not Correct")
+
+    @staticmethod
+    def shell_cmd_execute(cmd):
+        """
+         Execute Shell CMD
+        """
+        import subprocess
+        excute_result=subprocess.run(cmd,stdout=subprocess.PIPE)
+        return excute_result
+
+    def dd_execute(self,input,skip,count,output,seek,bs=512):
+        """
+        wrap the shell dd cmd
+        """
+        DD_DEFINE_BASE=["dd","iflag=skip_bytes,count_bytes","oflag=seek_bytes","conv=notrunc","status=none"]
+        DD_EXECUTE_CMD=["if={}".format(input),"skip={}".format(skip),"count={}".format(count),"of={}".format(output),"seek={}".format(seek),"bs={}".format(bs)]
+        self.shell_cmd_execute(DD_DEFINE_BASE+DD_EXECUTE_CMD)
+    def cpu_id_check(self):
+        """
+        CPU ID CHECK
+        """
+        cpu_device_id=[0x142fa,]
```

Comment:
We likely want to list all supported types here, ie. basic and advanced, possibly also variants like mine (0x140ff).

Reply:
this has  marked TBD,it would be filled when I get the correct CPUID of the basic and advanced board